"Out of gas issue" from brownie, in test_ve_ocean.py

NOTE: the test that's failing is skipped, as of commit https://github.com/oceanprotocol/ocean.py/commit/e42efcdb703540a4b209b8cf5706f8bd32b9b8b1. We will un-skip the test once we get it passing (and passing with clean changes)

----

### Overview of issue

To reproduce...

First, open `ocean_lib/models/vetest/test_ve_ocean.py` and comment out the `pytest.mark.skip()` line right before `test_ve_ocean1()`.

Then, have barge running.

Then, in a console, run:
```console
pytest ./ocean_lib/models/vetest/test_ve_ocean.py
```

The failure occurs when [`test_ve_ocean.py`](https://github.com/oceanprotocol/ocean.py/blob/main/ocean_lib/models/vetest/test_ve_ocean.py) makes a smart contract call to [`veOCEAN.vy::create_lock()`](https://github.com/oceanprotocol/contracts/blob/840c835df393917e99caa5dc292c8a47b25704f4/contracts/ve/veOCEAN.vy#L412):
```python
veOCEAN.create_lock(TA, t2, {"from": alice_wallet})
```

And the failure is:
```text
brownie.exceptions.VirtualMachineError: out of gas

# more detailed version
self = VirtualMachineError(ValueError("Gas estimation failed: 'VM Exception while processing transaction: revert Can only loc... in the future'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually."))
```

### Implementation of createLock():

From [veOCEAN.vy::createLock()](https://github.com/oceanprotocol/contracts/blob/840c835df393917e99caa5dc292c8a47b25704f4/contracts/ve/veOCEAN.vy#L412).

```python
@external
@nonreentrant('lock')
def create_lock(_value: uint256, _unlock_time: uint256):
    """
    @notice Deposit `_value` tokens for `msg.sender` and lock until `_unlock_time`
    @param _value Amount to deposit
    @param _unlock_time Epoch time when tokens unlock, rounded down to whole weeks
    """
    self.assert_not_contract(msg.sender)
    unlock_time: uint256 = (_unlock_time / WEEK) * WEEK  # Locktime is rounded down to weeks
    _locked: LockedBalance = self.locked[msg.sender]

    assert _value > 0  # dev: need non-zero value
    assert _locked.amount == 0, "Withdraw old tokens first"
    assert unlock_time > block.timestamp, "Can only lock until time in the future"
    assert unlock_time <= block.timestamp + MAXTIME, "Voting lock can be 4 years max"

    self._deposit_for(msg.sender, _value, unlock_time, _locked, CREATE_LOCK_TYPE)
```


### Full traceback

```text

config = {'ADDRESS_FILE': '~/.ocean/ocean-contracts/artifacts/address.json', 'DOWNLOADS_PATH': 'consume-downloads', 'METADATA_CACHE_URI': 'http://172.15.0.5:5000', 'NETWORK_NAME': 'development', ...}
factory_deployer_wallet = <LocalAccount '0xe2DD09d719Da89e5a3D0F2549c7E24566e947260'>, ocean_token = <ocean_lib.models.datatoken.Datatoken object at 0x7f94893137c0>
veOCEAN = <ocean_lib.models.ve_ocean.VeOcean object at 0x7f9488b11960>

    @pytest.mark.unit
    #@pytest.mark.skip(reason="Don't skip, once fixed #1096")
    def test_ve_ocean1(config, factory_deployer_wallet, ocean_token, veOCEAN):
        # inspiration from df-py/util/test/veOcean/test_lock.py
        assert veOCEAN.symbol() == "veOCEAN"
    
        OCEAN = ocean_token
    
        alice_wallet = accounts.add()  # new account avoids "withdraw old tokens first"
        factory_deployer_wallet.transfer(alice_wallet, "1 ether")
    
        TA = to_wei(0.0001)
        OCEAN.mint(alice_wallet.address, TA, {"from": factory_deployer_wallet})
    
        veOCEAN.checkpoint({"from": factory_deployer_wallet})
        OCEAN.approve(veOCEAN.address, TA, {"from": alice_wallet})
    
        t0 = chain.time()
        t1 = t0 // WEEK * WEEK + WEEK  # this is a Thursday, because Jan 1 1970 was
        t2 = t1 + WEEK
        chain.sleep(t1 - t0)
    
        assert OCEAN.balanceOf(alice_wallet.address) != 0
    
>       veOCEAN.create_lock(TA, t2, {"from": alice_wallet})

ocean_lib/models/vetest/test_ve_ocean.py:43: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
venv/lib/python3.10/site-packages/brownie/network/contract.py:1864: in __call__
    return self.transact(*args)
venv/lib/python3.10/site-packages/brownie/network/contract.py:1737: in transact
    return tx["from"].transfer(
venv/lib/python3.10/site-packages/brownie/network/account.py:682: in transfer
    receipt._raise_if_reverted(exc)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <Transaction '0x5f3e8151bbb65d7bf8140597c2571bb3b1ad8adefdb1fd5feb0b4b7fae01c380'>
exc = VirtualMachineError(ValueError({'message': 'VM Exception while processing transaction: out of gas', 'code': -32000, 'd... runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)', 'name': 'o'}}))

    def _raise_if_reverted(self, exc: Any) -> None:
        if self.status or CONFIG.mode == "console":
            return
        if not web3.supports_traces:
            # if traces are not available, do not attempt to determine the revert reason
            raise exc or ValueError("Execution reverted")
    
        if self._dev_revert_msg is None:
            # no revert message and unable to check dev string - have to get trace
            self._expand_trace()
        if self.contract_address:
            source = ""
        elif CONFIG.argv["revert"]:
            source = self._traceback_string()
        else:
            source = self._error_string(1)
            contract = state._find_contract(self.receiver)
            if contract:
                marker = "//" if contract._build["language"] == "Solidity" else "#"
                line = self._traceback_string().split("\n")[-1]
                if marker + " dev: " in line:
                    self._dev_revert_msg = line[line.index(marker) + len(marker) : -5].strip()
    
>       raise exc._with_attr(
            source=source, revert_msg=self._revert_msg, dev_revert_msg=self._dev_revert_msg
        )
E       brownie.exceptions.VirtualMachineError: out of gas

venv/lib/python3.10/site-packages/brownie/network/transaction.py:446: VirtualMachineError
------------------------------------------------------------------------------------------ Captured stdout setup ------------------------------------------------------------------------------------------
Transaction sent: 0x059179f0b3684be000f41ecb798bcb7b2286b4019fb18e2011d14643b5dbd1f0
  Gas price: 1e-09 gwei   Gas limit: 74437   Nonce: 392
  ERC20Template.mint confirmed   Block: 1139   Gas used: 67670 (90.91%)

Transaction sent: 0x037d9b245a3af936f50a5f9313e354e37e30a309720b84a1e10de04d16a73d20
  Gas price: 1e-09 gwei   Gas limit: 57923   Nonce: 393
  ERC20Template.mint confirmed   Block: 1140   Gas used: 52658 (90.91%)

Transaction sent: 0x81cfb5f60fd9043dab3740034812f5dcefe9284e9f26cdf2eb2eda5fc661ab8d
  Gas price: 1e-09 gwei   Gas limit: 57923   Nonce: 394
  ERC20Template.mint confirmed   Block: 1141   Gas used: 52658 (90.91%)

------------------------------------------------------------------------------------------ Captured stdout call -------------------------------------------------------------------------------------------
mnemonic: 'friend decrease rubber morning foster carpet sheriff nuclear assault lake order hat'
Transaction sent: 0xcfb467b0179a538d1e760803d659dbbb279b5b8b6405c2a373bfe3498bc6a480
  Gas price: 1e-09 gwei   Gas limit: 21000   Nonce: 395
  Transaction confirmed   Block: 1142   Gas used: 21000 (100.00%)

Transaction sent: 0xd324e2cab6b170c169b9782746eb22b3fd25e0764020c382e387d252304dd743
  Gas price: 1e-09 gwei   Gas limit: 57884   Nonce: 396
  ERC20Template.mint confirmed   Block: 1143   Gas used: 52622 (90.91%)

Transaction sent: 0x848d1c84618c0ed7bb29dbcecbcf6e25612267972e151405d437381ba2ae89e8
  Gas price: 1e-09 gwei   Gas limit: 94426   Nonce: 397
  veOCEAN.checkpoint confirmed   Block: 1144   Gas used: 85842 (90.91%)

Transaction sent: 0xb5e40639e817fbfb7cb06342e7c5a820f5ed95a3dafaa93ebac88990c4976a86
  Gas price: 1e-09 gwei   Gas limit: 48583   Nonce: 0
  ERC20Template.approve confirmed   Block: 1145   Gas used: 44167 (90.91%)

Transaction sent: 0x5f3e8151bbb65d7bf8140597c2571bb3b1ad8adefdb1fd5feb0b4b7fae01c380
  Gas price: 1e-09 gwei   Gas limit: 410637   Nonce: 1
  veOCEAN.create_lock confirmed (out of gas)   Block: 1146   Gas used: 410637 (100.00%)

========================================================================================= short test summary info =========================================================================================
FAILED ocean_lib/models/vetest/test_ve_ocean.py::test_ve_ocean1 - brownie.exceptions.VirtualMachineError: out of gas
=========================================================================================== 1 failed in 30.83s ============================================================================================

```

### CI failure

Example similar CI failure here: (it's called test1 there).
https://github.com/oceanprotocol/ocean.py/actions/runs/3533298422/jobs/5928743050#step:7:137

It occurred for ve contracts when creating the lock.

--- a/ocean_lib/models/vetest/test_ve_ocean.py
+++ b/ocean_lib/models/vetest/test_ve_ocean.py
@@ -18,6 +18,7 @@ MAXTIME = 4 * 365 * 86400  # 4 years
 
 
 @pytest.mark.unit
+@pytest.mark.skip(reason="Don't skip, once fixed #1096")
 def test_ve_ocean1(config, factory_deployer_wallet, ocean_token, veOCEAN):
     # inspiration from df-py/util/test/veOcean/test_lock.py
     assert veOCEAN.symbol() == "veOCEAN"

--- a/ocean_lib/models/vetest/test_ve_ocean.py
+++ b/ocean_lib/models/vetest/test_ve_ocean.py
@@ -4,6 +4,7 @@
 #
 import brownie
 import pytest
+from web3 import Web3
 
 from ocean_lib.models.ve_ocean import VeOcean
 from ocean_lib.ocean.mint_fake_ocean import mint_fake_OCEAN
@@ -17,7 +18,6 @@ MAXTIME = 4 * 365 * 86400  # 4 years
 
 
 @pytest.mark.unit
-@pytest.mark.skip(reason="Don't skip, once fixed #1096")
 def test_ve_ocean1(config, factory_deployer_wallet, ocean_token, veOCEAN):
     # inspiration from df-py/util/test/veOcean/test_lock.py
     assert veOCEAN.symbol() == "veOCEAN"
@@ -27,7 +27,7 @@ def test_ve_ocean1(config, factory_deployer_wallet, ocean_token, veOCEAN):
     alice_wallet = accounts.add()  # new account avoids "withdraw old tokens first"
     factory_deployer_wallet.transfer(alice_wallet, "1 ether")
 
-    TA = to_wei(0.0001)
+    TA = Web3.toWei(0.0001, "ether")
     OCEAN.mint(alice_wallet.address, TA, {"from": factory_deployer_wallet})
 
     veOCEAN.checkpoint({"from": factory_deployer_wallet})
@@ -49,8 +49,8 @@ def test_ve_ocean1(config, factory_deployer_wallet, ocean_token, veOCEAN):
 
     assert veOCEAN.get_last_user_slope(alice_wallet) != 0
 
-    alice_vote_power = from_wei(veOCEAN.balanceOf(alice_wallet, chain.time()))
-    expected_vote_power = from_wei(TA) * WEEK / MAXTIME
+    alice_vote_power = float(Web3.fromWei(veOCEAN.balanceOf(alice_wallet, chain.time()), "ether"))
+    expected_vote_power = float(Web3.fromWei(TA, "ether")) * WEEK / MAXTIME
     assert alice_vote_power == pytest.approx(expected_vote_power, TA / 20.0)
 
     brownie.network.chain.sleep(t2)
@@ -61,11 +61,3 @@ def test_ve_ocean1(config, factory_deployer_wallet, ocean_token, veOCEAN):
 
     assert veOCEAN.get_last_user_slope(alice_wallet) == 0
     assert veOCEAN.balanceOf(alice_wallet, chain.time()) == 0
-
-
-def to_wei(amt_eth) -> int:
-    return int(amt_eth * 1e18)
-
-
-def from_wei(amt_wei: int) -> float:
-    return float(amt_wei / 1e18)

--- a/ocean_lib/models/vetest/test_ve_ocean.py
+++ b/ocean_lib/models/vetest/test_ve_ocean.py
@@ -49,7 +49,9 @@ def test_ve_ocean1(config, factory_deployer_wallet, ocean_token, veOCEAN):
 
     assert veOCEAN.get_last_user_slope(alice_wallet) != 0
 
-    alice_vote_power = float(Web3.fromWei(veOCEAN.balanceOf(alice_wallet, chain.time()), "ether"))
+    alice_vote_power = float(
+        Web3.fromWei(veOCEAN.balanceOf(alice_wallet, chain.time()), "ether")
+    )
     expected_vote_power = float(Web3.fromWei(TA, "ether")) * WEEK / MAXTIME
     assert alice_vote_power == pytest.approx(expected_vote_power, TA / 20.0)
 

